rskj-core/src/main/java/co/rsk/rpc/modules/personal/PersonalModuleWalletEnabled.java

```diff
@@ -193,7 +195,14 @@ private String sendTransaction(Web3.CallArguments args, Account account) throws
         BigInteger accountNonce = args.nonce != null ? TypeConverter.stringNumberAsBigInt(args.nonce) : transactionPool.getPendingState().getNonce(account.getAddress());
         BigInteger value = args.value != null ? TypeConverter.stringNumberAsBigInt(args.value) : BigInteger.ZERO;
         BigInteger gasPrice = args.gasPrice != null ? TypeConverter.stringNumberAsBigInt(args.gasPrice) : BigInteger.ZERO;
-        BigInteger gasLimit = args.gas != null ? TypeConverter.stringNumberAsBigInt(args.gas) : BigInteger.valueOf(GasCost.TRANSACTION);
+
+        BigInteger gasLimit = DEFAULT_GAS_LIMIT;
+        if(args.gasLimit != null) {
+        	gasLimit = TypeConverter.stringNumberAsBigInt(args.gasLimit);
+        }
+        if(args.gas != null) {
+        	gasLimit = TypeConverter.stringNumberAsBigInt(args.gas);
+        }
```

Comment:
looks like this duplicates the code above. As an option this can be extracted in some utility method..

Reply:
Sorry, what code does it duplicate?

Reply:
I meant this part
```
        BigInteger gasLimit = DEFAULT_GAS_LIMIT;
        if(args.gasLimit != null) {
        	gasLimit = TypeConverter.stringNumberAsBigInt(args.gasLimit);
        }
        if(args.gas != null) {
        	gasLimit = TypeConverter.stringNumberAsBigInt(args.gas);
        }
```

[here](https://github.com/rsksmart/rskj/pull/1520/files#diff-445274bc5d69cec4531468ad84ff8367279dd9230c358cd6d664d41d5c781afbR68) and [here](https://github.com/rsksmart/rskj/pull/1520/files#diff-8f7264e434eef578b025bfda335f71ba4bcb69ccf464722a11a0d1cab1af88c2R199)

Reply:
Yes, I see. 
But the whole method is repeated. I think this specific situation needs a more careful refactor. If you agree, we let this for another moment and I will move on with this topic.